tests/test_dse.py

```diff
@@ -163,82 +162,20 @@ def test_cse(exprs, expected):
     ('3*fa[x]**4', '3*(fa[x]*fa[x]*fa[x]*fa[x])'),
     ('fa[x]**2', 'fa[x]*fa[x]'),
     ('1/(fa[x]**2)', '1/(fa[x]*fa[x])'),
+    ('1/(fb[x]**2 + 1)', '1/(fb[x]*fb[x] + 1)'),
     ('1/(fa[x] + fb[x])', '1/(fa[x] + fb[x])'),
     ('3*sin(fa[x])**2', '3*(sin(fa[x])*sin(fa[x]))'),
+    ('fa[x]/(fb[x]**2)', 'fa[x]/((fb[x]*fb[x]))')
 ])
 def test_pow_to_mul(expr, expected):
     grid = Grid((4, 5))
     x, y = grid.dimensions
+
+    s = Scalar(name='s')  # noqa
     fa = Function(name='fa', grid=grid, dimensions=(x,), shape=(4,))  # noqa
     fb = Function(name='fb', grid=grid, dimensions=(x,), shape=(4,))  # noqa
-    assert str(pow_to_mul(eval(expr))) == expected
-
-
-@pytest.mark.parametrize('exprs,expected', [
-    # none (different distance)
-    (['Eq(t0, fa[x] + fb[x])', 'Eq(t1, fa[x+1] + fb[x])'],
-     {'fa[x] + fb[x]': 'None', 'fa[x+1] + fb[x]': 'None'}),
-    # none (different dimension)
-    (['Eq(t0, fa[x] + fb[x])', 'Eq(t1, fa[x] + fb[y])'],
-     {'fa[x] + fb[x]': 'None', 'fa[x] + fb[y]': 'None'}),
-    # none (different operation)
-    (['Eq(t0, fa[x] + fb[x])', 'Eq(t1, fa[x] - fb[x])'],
-     {'fa[x] + fb[x]': 'None', 'fa[x] - fb[x]': 'None'}),
-    # simple
-    (['Eq(t0, fa[x] + fb[x])', 'Eq(t1, fa[x+1] + fb[x+1])', 'Eq(t2, fa[x-1] + fb[x-1])'],
-     {'fa[x] + fb[x]': 'Stencil([(x, {-1, 0, 1})])'}),
-    # 2D simple
-    (['Eq(t0, fc[x,y] + fd[x,y])', 'Eq(t1, fc[x+1,y+1] + fd[x+1,y+1])'],
-     {'fc[x,y] + fd[x,y]': 'Stencil([(x, {0, 1}), (y, {0, 1})])'}),
-    # 2D with stride
-    (['Eq(t0, fc[x,y] + fd[x+1,y+2])', 'Eq(t1, fc[x+1,y+1] + fd[x+2,y+3])'],
-     {'fc[x,y] + fd[x+1,y+2]': 'Stencil([(x, {0, 1}), (y, {0, 1})])'}),
-    # 2D with subdimensions
-    (['Eq(t0, fc[xi,yi] + fd[xi+1,yi+2])', 'Eq(t1, fc[xi+1,yi+1] + fd[xi+2,yi+3])'],
-     {'fc[xi,yi] + fd[xi+1,yi+2]': 'Stencil([(xi, {0, 1}), (yi, {0, 1})])'}),
-    # 2D with constant access
-    (['Eq(t0, fc[x,y]*fc[x,0] + fd[x,y])', 'Eq(t1, fc[x+1,y+1]*fc[x+1,0] + fd[x+1,y+1])'],
-     {'fc[x,y]*fc[x,0] + fd[x,y]': 'Stencil([(x, {0, 1}), (y, {0, 1})])'}),
-    # 2D with multiple, non-zero, constant accesses
-    (['Eq(t0, fc[x,y]*fc[x,0] + fd[x,y]*fc[x,1])',
-      'Eq(t1, fc[x+1,y+1]*fc[x+1,0] + fd[x+1,y+1]*fc[x+1,1])'],
-     {'fc[x,0]*fc[x,y] + fc[x,1]*fd[x,y]': 'Stencil([(x, {0, 1}), (y, {0, 1})])'}),
-    # 2D with different shapes
-    (['Eq(t0, fc[x,y]*fa[x] + fd[x,y])', 'Eq(t1, fc[x+1,y+1]*fa[x+1] + fd[x+1,y+1])'],
-     {'fc[x,y]*fa[x] + fd[x,y]': 'Stencil([(x, {0, 1}), (y, {0, 1})])'}),
-    # complex (two 2D aliases with stride inducing relaxation)
-    (['Eq(t0, fc[x,y] + fd[x+1,y+2])', 'Eq(t1, fc[x+1,y+1] + fd[x+2,y+3])',
-      'Eq(t2, fc[x-2,y-2]*3. + fd[x+2,y+2])', 'Eq(t3, fc[x-4,y-4]*3. + fd[x,y])'],
-     {'fc[x,y] + fd[x+1,y+2]': 'Stencil([(x, {-1, 0, 1}), (y, {-1, 0, 1})])',
-      '3.*fc[x-3,y-3] + fd[x+1,y+1]': 'Stencil([(x, {-1, 0, 1}), (y, {-1, 0, 1})])'}),
-])
-def test_collect_aliases(exprs, expected):
```

Comment:
@mloubout this was *not* removed -- just moved inside the TestAliases class, where it should always have been